src/TRestTrackLineAnalysisProcess.cxx

```diff
@@ -0,0 +1,252 @@
+
+//////////////////////////////////////////////////////////////////////////
+/// The TRestTrackLineAnalysis process is meant to analyze lienar tracks,
+/// such as alpha tracks. It should retreive the information after performing
+/// TRestTrackLinearization process, for the time being only works when
+/// XZ and YZ are present in the readout. Moreover, the hits have to be stored
+/// following the minimum path. It assumes that only one alpha track
+/// is present, using the parameter fTrackBalance to define the minimun energy
+/// fTrackBalance*totalEnergy of the total XZ or YZ energy to assume a single
+/// track event, if the condition is not fulfilled the event is rejected.
+/// Several parameters of the alphaTrack are extracted such as origin, end,
+/// length, energy and downwards (bool).
+/// The different parameters required by this process are:
+/// * fTrackBalance : Minimum energy balance to consider single track event,
+/// (see above for more details.)
+///
+/// ### Observables
+///
+/// * **originX**: Track origin in the X cordinate
+/// * **originY**: Track origin in the Y cordinate
+/// * **originZ**: Track origin in the Z cordinate
+/// * **endX**: Track end in the X cordinate
+/// * **endY**: Track end in the Y cordinate
+/// * **endZ**: Track end in the Z cordinate
+/// * **length**: Track length in mm
+/// * **angle**: Track polar angle in radians
+/// * **downwards**: (bool) true if the track direction is downwards, false otherwise
+/// * **totalEnergy**: Energy of the track
+///
+/// Metadata example for this process:
+/// \code
+///            <addProcess type="TRestTrackLineAnalysisProcess" name="alphaTrackAna" value="ON"
+///            verboseLevel="silent" >
+///                   <parameter name="trackBalance" value="0.75" />
+///                   <observable name="originX" value="ON" />
+///                   <observable name="originY" value="ON" />
+///                   <observable name="originZ" value="ON" />
+///                   <observable name="endX" value="ON" />
+///                   <observable name="endY" value="ON" />
+///                   <observable name="endZ" value="ON" />
+///                   <observable name="length" value="ON" />
+///                   <observable name="angle" value="ON" />
+///                   <observable name="downwards" type="bool" value="ON" />
+///                   <observable name="totalEnergy" value="ON" />
+/// 		</addProcess>
+/// \endcode
+///
+///--------------------------------------------------------------------------
+///
+/// RESTsoft - Software for Rare Event Searches with TPCs
+///
+/// History of developments:
+///
+/// 2022-January First implementation
+///
+///              JuanAn Garcia
+///
+///
+///
+/// <hr>
+///
+
+#include "TRestTrackLineAnalysisProcess.h"
+#include "TRestTrackReductionProcess.h"
+using namespace std;
+
+ClassImp(TRestTrackLineAnalysisProcess);
+
+//______________________________________________________________________________
```

Comment:
Not the good formatting. Using `REST_MakeProcess` will produce the standard formatting.